### PR TITLE
Settings page

### PR DESCRIPTION
**Pull request by Velenir**
Seemingly works
`/settings`
But needs testing

https://pr1160--dexreact.review.gnosisdev.com/settings

localhost_8080_settings (1)

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr1160--dexreact.review.gnosisdev.com

File: src/pages/Settings.tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+const SettingsButtonReset = styled(SettingsButtonSubmit)`
+  background: #c7cbda;
+  color: var(--color-text-active);
+`
+
+export interface SettingsFormData {
+  walletconnect: WCOptions
+}
+
+export interface ResolverResult<T extends SettingsFormData, K extends keyof T = keyof T> {
```
[the remark this answers is not in the document]
Comment (Velenir) [reply src/pages/Settings.tsx:38]: No, it's `K = keyof T` as default

File: src/utils/walletconnectOptions.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+export const getWCOptionsFromStorage = (): WCOptions => {
+  const storedOptions = localStorage.getItem(STORAGE_KEY_CUSTOM_WC_OPTIONS)
+  if (!storedOptions) return {}
+
+  const { infuraId, bridge, rpc }: WCOptions = JSON.parse(storedOptions)
+  return {
+    infuraId,
+    bridge,
+    rpc,
+  }
```
[the remark this answers is not in the document]
Comment (Velenir) [reply src/utils/walletconnectOptions.ts:34]: I must have been tired 😴

**Comment by Velenir** [comment]
Did you remove id by clearing input and clocking Apply?
It should reprompt when you have ongoing WC connection and change WC settings, for you to re-establish connection with new settings.
If you had connection with custom id and reset to default id then it should reprompt.
Was that it?

**Comment by alfetopito** [comment]
> Did you remove id by clearing input and clicking Apply?

Yup

> It should reprompt when you have ongoing WC connection and change WC settings, for you to re-establish connection with new settings. And then redirect to home.
> If no ongoing WC then no reprompt.
> If you had connection with custom id and reset to default id then it should reprompt.
> Was that it?

Yep. Then is working as expected.

File: src/pages/Settings.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (alfetopito) [reply src/pages/Settings.tsx:38]: I guess I understood...

**Comment by W3stside** [comment]
Unpopular opinion: make inputs expand on focus OR stack as column on smaller size and + add a larger margin between the inputs and the `OR` text and align the titles (`InfuraId` etc) horizontally with the inputs :)

also less `box-shadow` - is that the `var(--box-shadow)` amount? seems hefty but anyways less would be kew

**Comment by Velenir** [comment]
> Unpopular opinion: make inputs expand on focus OR stack as column on smaller size and + add a larger margin between the inputs and the `OR` text and align the titles (`InfuraId` etc) horizontally with the inputs :)


Honestly, I'd rather not do too much with the styling :wink: , on account of this here being a _secret_ settings page.
But anyone is welcome to make it prettier :wink: :wink: .

Inputs already stack vertically when on Mobile size. Or I don't quite understand what you mean.

**Pull request merged**